lib/internal/streams/async_iterator.js

```diff
@@ -110,17 +110,27 @@ const ReadableStreamAsyncIteratorPrototype = Object.setPrototypeOf({
   },
 
   return() {
-    // destroy(err, cb) is a private API.
-    // We can guarantee we have that here, because we control the
-    // Readable class this is attached to.
     return new Promise((resolve, reject) => {
-      this[kStream].destroy(null, (err) => {
-        if (err) {
+      const stream = this[kStream];
+
+      // TODO(ronag): Remove this check once finished() handles
+      // already destroyed streams.
```

Comment:
Isn’t that part of another PR?

Reply:
yes, put this as blocked and wait for that first?

Reply:
https://github.com/nodejs/node/pull/28748

Reply:
it's not exactly the same, #28748 does not check readableEnded.